andela/travel_tool_back
# #161744068 Fix guest room change issue

#### What does this PR do?
Fixes the bug that exists when an admin attempts to change the room of a requester

#### Description of Task to be completed?
- Add validation to available room checker to ignore rejected requests
- Change the error message displayed when the opposite gender already booked the room

#### How should this be manually tested?
Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
Enter into the project directory - `cd travel_tool_back`
checkout the branch `git checkout bg-change-guest-room-161744068`
Install dependencies using `yarn install`
Rollback and run database migrations with `yarn db:rollmigrate`
Start the server with `make start`
Visit https://travela-local.andela.com
Login to the application
Ensure you have a travel admin privilege
Navigate to the residence manage page and click on a guesthouse with bookings
Click on any booking record and try to change the allocated room to another
You should be able to change the booking record from one room to another

#### Any background context you want to provide?
- You should have a few request on the system so you can view the booked rooms for the requests.
- Once a room is booked, it becomes unavailable whether the manager has approved it or not.

#### What are the relevant pivotal tracker stories?
[#161744068](https://www.pivotaltracker.com/story/show/161744068)

#### Screenshots (if appropriate)

#### Questions:
